### DISCUSSION TITLE
Argument List Too Long Error

### DISCUSSION THREAD

**Discussion by uriel-tannen** [ImageMagick/ImageMagick#2990] [General]
I've got the following code that calculates the average RGB value for a list of images in a directory, and then outputs these values to a CSV file. This works well up to around ~250 images, but anything more than this gives me an `argument list too long` error. 

`magick *.jpg -format "%f,%[fx:int(mean.r*255)],%[fx:int(mean.g*255)],%[fx:int(mean.b*255)]\n" info: > swatchesRGB.csv`

Can I restructure this command to avoid this error message? I have 20k+ images to do, so batch processing these all at once would be most helpful. 

I'm using IM7 on a Windows PC.

Thank you

**Comment by snibgo** [comment]
You command reads _all_ your jpg files. Then it processes each one, calculating the means and writing them to stdout. If your images are large (with many pixels), this will eat memory.

Instead, I suggest you run magick for one file at a time. Put this inside a shell "for" loop, like this:
```
( for %F in (xr*.jpg) do magick %F -format %f,%[fx:mean]\n info: ) >s.csv
```
If you put this in a BAT script, double each percent sign %.

**Reply by uriel-tannen** [reply]
Thanks @snibgo. I've slightly reformatted what you suggested to this:

```
for filename in *.jpg; do 
    magick $filename -format %f,%[fx:mean]\n info: >s.csv
done
```

However for each image in the folder, it's replacing the contents of s.csv. As in, if I run this on X images rather than having X lines of data in the CSV file, I always have just the single line - which is the last image processed. How do I change this so that, by putting the command into a loop, it adds to the CSV rather than replace what's in it?

**Comment by snibgo** [comment]
You use a Windows PC, so I showed you a Windows command. But I now guess you are using bash, so of course my command is wrong for you.

">s.csv" will overwrite s.csv. ">>s.csv" will append to csv. So you could start by deleting s.csv, then use ">>". Alternatively, for better performance, do as I did: use parentheses around the entire "for" command, and redirect the entire contents, using "s.csv".

This is fairly basic shell scripting. I suggest you find a web page that teaches these things.
